docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+Four routing strategies are currently supported:
+
+-   `networkFirst` attempts to load a resource over the network, falling back on the cache if the request fails or times out. This is a useful strategy for assets that either change frequently or may change frequently (i.e., are not versioned).
+-   `cacheFirst` loads a resource from the cache unless it does not exist, in which case it is fetched over the network. This is best for resources that change infrequently or can be cached for a long time (e.g., versioned assets).
+-   `networkOnly` forces the resource to always be retrieved over the network, and is useful for requests that have no offline equivalent.
+-   `staleWhileRevalidate` requests resources from both the cache and the network simulaneously. The cache is updated with each successful network response. This strategy is best for resources that do not need to be continuously up-to-date, like user avatars. However, when fetching third-party resources that do not send CORS headers, it is not possible to read the contents of the response or verify the status code. As such, it is possible that a bad response could be cached. In such cases, the `networkFirst` strategy may be a better fit.
+
+# Conditional code
+
+Webpack's static code analyzer is capable of removing dead code branch from the bundles it creates. The Dojo build tool leverages this capability to create branches of conditional code that can be removed at build time. Named conditional blocks are defined using Dojo framework's `has` module. Then those conditional blocks can be statically set to true or false through `.dojorc` and removed at build time.
+
+> main.ts
+
+```ts
+import has from '@dojo/framework/has';
+
+if (has('production')) {
+	console.log('Starting in production');
+} else {
+	console.log('Starting in dev mode');
+}
+
+export const mode = has('production') ? 'dist' : 'dev';
+```
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"features": {
+			"production": true
+		}
+	}
+}
+```
+
+The above `production` feature will be set `true` for **production builds** (`dist` mode). The Dojo build system uses the [static-build-loader](https://github.com/dojo/webpack-contrib/#static-build-loader) to rewrite `has()` calls that have been statically asserted by the system as `true` or `false`. This allows Webpack to conditionally identify code as unreachable and remove those dead code branches from the build.
+
+For example, the above code would be rewritten as:
+
+> static-build-loader output
+
+```js
+import has from '@dojo/framework/has';
+
+if (true) {
+	console.log('Starting in production');
+} else {
+	console.log('Starting in dev mode');
+}
+
+export const mode = true ? 'dist' : 'dev';
+```
+
+Webpack's dead branch removal would then remove the unreachable code.
```

Comment:
This seems to be too low level to explain in a reference guide and it's not actually webpack that does this anyway.